api/app/hfi/hfi_request.py

```diff
@@ -0,0 +1,38 @@
+""" HFI request logic """
+
+from itertools import groupby
+import operator
+from typing import List
+from app.db.models.hfi_calc import PlanningWeatherStation
+from app.schemas.hfi_calc import HFIResultRequest, StationInfo
+
+
+def update_result_request(result_request: HFIResultRequest,
+                          latest_stations: List[PlanningWeatherStation]):
+    """ Returns copy of the result request with the latest planning weather stations """
+    if len(latest_stations) == 0:
+        return result_request
+
+    updated_result_request = result_request.copy(deep=True)
+    get_attr = operator.attrgetter('planning_area_id')
+    stations_by_planning_area = dict((k, list(map(lambda x: x, values)))
+                                     for k, values in groupby(latest_stations, get_attr))
+    for planning_area_id in updated_result_request.planning_area_station_info:
+        existing_stations = result_request.planning_area_station_info[planning_area_id]
```

Comment:
also, what are "existing stations"?

Reply:
Stations that were persisted in the last stored request

Reply:
`request_stations`?